### PR TITLE
detect/flowbits: use strtok_r for parsing 

### PR DESCRIPTION
**Pull request by inashivb**
Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:
https://redmine.openinfosecfoundation.org/issues/5154

Previous PR: https://github.com/OISF/suricata/pull/7196

Changes since v1:
- Special handling for flowbits OR op with spaces
- Test for bug 5154
- Removal of dead pcre regex code

### CODE REVIEW

**Comment by inashivb** [comment]
@catenacyber could you please tell if this fuzzing error is for me to solve?

File: src/detect-flowbits.c [hunk 1]
```diff
@@ -212,13 +207,124 @@ int DetectFlowbitMatch (DetectEngineThreadCtx *det_ctx, Packet *p,
...
+                *val++ = '\0';
+                while (*val != '\0' && isblank(*val)) {
+                    val++;
+                }
+            } else {
+                SCLogDebug("val %s", token);
+            }
+        }
+        if (strlen(token) == 0) {
+            goto next;
```
Comment (victorjulien) [inline src/detect-flowbits.c:239]: should this be an error? It suggests an empty token, right?

File: src/detect-flowbits.c [hunk 2]
```diff
@@ -212,13 +207,124 @@ int DetectFlowbitMatch (DetectEngineThreadCtx *det_ctx, Packet *p,
...
+        DetectEngineCtx *de_ctx, const char *rawstr, DetectFlowbitsData **cdout)
 {
+    bool cmd_set = false;
+    bool name_set = false;
+    uint8_t cmd = 0;
+    char name[256] = "";
+    char *val = NULL;
+    DetectFlowbitsData *cd = NULL;
+    char copy[strlen(rawstr) + 1];
+    strlcpy(copy, rawstr, sizeof(copy));
```
Comment (victorjulien) [inline src/detect-flowbits.c:220]: wonder if things would be simpler during parsing if we would copy only the non-`is_blank()` characters here. Then during tokenizing below we could ignore any spaces related logic.

File: src/detect-flowbits.c [hunk 3]
```diff
@@ -212,13 +207,124 @@ int DetectFlowbitMatch (DetectEngineThreadCtx *det_ctx, Packet *p,
...
+                "idx %" PRIu32 ", cmd %d, name %s", cd->idx, cmd, strlen(name) ? name : "(none)");
+    }
+    *cdout = cd;
+
+    return 0;
+error:
+    if (cd != NULL)
+        DetectFlowbitFree(de_ctx, cd);
+    return -1;
+#if 0
```
Comment (victorjulien) [inline src/detect-flowbits.c:322]: remove

**Review by victorjulien — CHANGES_REQUESTED** [review]

**Comment by suricata-qa** [comment]
Information: QA ran without warnings. 

Pipeline 6768

File: src/detect-flowbits.c (hunk 2 above)
Comment (inashivb) [reply src/detect-flowbits.c:220]: How do we handle "key<space>value" pairs then?
e.g. `set flowbit1`.

**Pull request closed without merging**